deftree/svcparse/parser.go

```diff
@@ -305,7 +360,7 @@ func ParseHttpBindings(lex *SvcLexer) ([]*HTTPBinding, error) {
 				rv = append(rv, more_bindings...)
 				good_position = lex.GetPosition()
 			} else if tk == EOF || tk == ILLEGAL {
-				return nil, parseErr("non-illegal token while parsing HttpBindings", lex.GetLineNumber(), fmt.Sprintf("(%v) of type %v", val, tk))
+				return nil, parseErr("legal token while parsing HttpBindings", lex.GetLineNumber(), fmt.Sprintf("(%v) of type %v", val, tk))
```

Comment:
It seems like this error message should be indicating that EOF or an Illegal token is expected? Perhaps the parser is looking to finish its work. "legal token" doesn't tell me much and I might just as quickly assume that this was a typo and that illegal was meant here.

Reply:
That line will actually generate an error with the value:

    parser expected legal token while parsing HttpBindings in line '##', instead found '(BAD_THING) of type EXAMPLE'

which I feel is a relatively descriptive error message.

Reply:
As to your review question about whether the comments make their way into the docs: No, no they do not. You see, I forgot to check how this affects documentation generation and it is only because of your comment that I'm checking now. Unfortunately, the docs generation for this is pretty broken, something I had forgotten about.

So I'm going to extend documentation generation to support this new addition, push the changes, and update this PR when I do.

Reply:
@hasLeland I think @zaquestion has a point about the code being confusing even if it produces a good error message. `parseErr` as a function also implies to me that is going to parse an error rather than return a error.

Maybe a rename to `parserErr` the extra `r` makes a difference to me in understanding what it does.

-------------------------------------------------------------
Extra:

I wrote this out for making a type for parse error, so I am going to post it, though it may be more than we need.
```
type parseErr struct {
	expected string 
	line int 
	val string
}

func (p parseErr) Error() string {
    return fmt.Errorf("parser expected %v in line '%v', instead found '%v'", p.expected, p.line, p.val)
}

...
return nil, parseErr {
   expected: "legal token while parsing HttpBindings", 
   line: lex.GetLineNumber(), 
   val: fmt.Sprintf("(%v) of type %v", val, tk)),
}
```